projects/plugins/crm/includes/ZeroBSCRM.Permissions.php

```diff
@@ -692,13 +692,49 @@ function zeroBSCRM_permsObjType($objTypeID=-1){
 		return false;
 	}
 
-	function zeroBSCRM_permsCustomers(){
+/**
+ * Determine if a user is allowed to manage contacts.
+ *
+ * @since TBD
+ *
+ * @param WP_User $user The WP User to check permission access for.
+ * @return bool Returns a bool representing a user permission state.
+ */
+function jpcrm_can_user_manage_contacts( WP_User $user ) {
+	/**
+	 * Allow third party plugins to modify the permission conditions for contacts.
+	 *
+	 * @since TBD
+	 *
+	 * @param boolean $allowed A boolean that represents the permission state.
+	 * @param WP_User $user The WP User to check permission access for.
+	 */
+	return (bool) apply_filters(
+		'jpcrm_can_user_manage_contacts',
+		$user->has_cap( 'admin_zerobs_customers' ),
+		$user
+	);
+}
 
-	    $cu = wp_get_current_user();
-	    if ($cu->has_cap('admin_zerobs_customers')) return true;
-	    return false;
+/**
+ * Determine if the current user is allowed to manage contacts.
+ *
+ * @deprecated TBD Use jpcrm_can_user_manage_contacts()
+ *
+ * @return bool
+ *
+ * @phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+ */
+function zeroBSCRM_permsCustomers() {
+	$current_user = wp_get_current_user();
+
+	if ( ! $current_user instanceof WP_User ) {
+		return false;
 	}
 
+	return jpcrm_can_user_manage_contacts( $current_user ) === true;
+}
```

Comment:
```suggestion
 * @deprecated $$next-version$$ Use jpcrm_can_user_manage_contacts()
 *
 * @return bool
 *
 * @phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
 */
function zeroBSCRM_permsCustomers() {
	_deprecated_function( __FUNCTION__, 'crm-$$next-version$$' );
	$current_user = wp_get_current_user();

	if ( ! $current_user instanceof WP_User ) {
		return false;
	}

	return jpcrm_can_user_manage_contacts( $current_user ) === true;
}
```

Reply:
Just a suggestion, no harm in adding this in just now given the deprecation suggestions :)

Reply:
> That said, do we really need to introduce another helper function + filter? Why not just use the native has_cap() WP function directly?

I was wondering the same... I was trying to see a use case for allowing third parties to change the user permissions.

Reply:
> do we really need to introduce another helper function + filter? Why not just use the native has_cap() WP function directly?

**Filter**

This is just my own opinion about plugin development, but In general I tend to lean towards making our systems "Integrable" even in situations where we don't currently have a need for the hook(s) ourselves.

It's kind of a judgement call when to add hooks if that's the case, but to give this one some specific context, then we'd be able to create an extension (e.g.: "Granular Team and User Permissions") by extending the filter - or someone else can come up with other types of ideas. If we never open up to these extensions/modifications, we never really allow the community to get involved in "the extension market".

**Function**

If I remember correctly, then we have ~50 instances of `zeroBSCRM_permsCustomers` in core right now so it's definitely something we're repeating a lot which makes me immediately think it's supposed to follow the DRY principle and therefor make it easier to maintain (e.g.: especially if we keep the filter around - otherwise it would be a potential nightmare to update it in the future 😄)

Reply:
>If we do deprecate this function, at minimum let's update all calls to said function in core and make an issue in the extensions repo to update the various extensions that use said call.

I'm leaning towards this as well, but I'd prefer to keep it as a separate issue for two reasons:

* A separate PR makes sense since the old function is used in ~50 places in the code and it will take away focus from the main objective here.
* If we add `_deprecated_function( __FUNCTION__, 'crm-$$next-version$$' );` in this PR, then we'd be flooding our users error logs with warning since it's used in so many spaces, so the condition would (IMO) belong in the separate PR in this specific instance 🗒️
  * If it was a smaller change to replace all the old uses, then I wouldn't mind adding `_deprecated_function` in this PR just to reduce overhead.

 (I'll create an issue depending on the feedback about keeping the new permission check function)

Reply:
> A separate PR makes sense since the old function is used in ~50 places in the code and it will take away focus from the main objective here.

I Agree.